[Private Network] NullPointerException on NodeSmartContractV2PermissioningController.createPayload

### Description
Besu is throwing NullpointerException in all other nodes connected when some other node is removed from the permissioning smart contract v2 (If I have nodes n1, n2, n3 and n4, and then remove n4, the nodes n1, n2 and n3 throw the exception).

### Steps to Reproduce (Bug)
1. Start a private network
2. Deploy permissioning smart contract v2 defining nodeRulesContract.activateValidationEnodeIdOnly(true) 
3. Remove a node from the smart contract

**Expected behavior:** 
* Node removed

**Actual behavior:**
* NullpointerException

**Frequency:** Everytime a node is removed and then when the nodes are restarted

### Logs
```
2023-04-19T14:22:02.024-0300 ERROR State machine threw exception while processing event \{org.hyperledger.besu.consensus.common.bft.events.BftReceivedMessageEvent@5c2fb8e5\}
java.lang.RuntimeException: Error building payload to call node permissioning smart contract
	at org.hyperledger.besu.ethereum.permissioning.NodeSmartContractV2PermissioningController.createPayload(NodeSmartContractV2PermissioningController.java:110) ~[besu-permissioning-23.4.0-SNAPSHOT.jar:23.4.0-dev-d54a1bf2]
	at org.hyperledger.besu.ethereum.permissioning.NodeSmartContractV2PermissioningController.getCallResult(NodeSmartContractV2PermissioningController.java:81) ~[besu-permissioning-23.4.0-SNAPSHOT.jar:23.4.0-dev-d54a1bf2]
	at org.hyperledger.besu.ethereum.permissioning.NodeSmartContractV2PermissioningController.isPermitted(NodeSmartContractV2PermissioningController.java:69) ~[besu-permissioning-23.4.0-SNAPSHOT.jar:23.4.0-dev-d54a1bf2]
	at org.hyperledger.besu.ethereum.permissioning.NodeSmartContractV2PermissioningController.checkSmartContractRules(NodeSmartContractV2PermissioningController.java:60) ~[besu-permissioning-23.4.0-SNAPSHOT.jar:23.4.0-dev-d54a1bf2]
	at org.hyperledger.besu.ethereum.permissioning.AbstractNodeSmartContractPermissioningController.isConnectionPermitted(AbstractNodeSmartContractPermissioningController.java:87) ~[besu-permissioning-23.4.0-SNAPSHOT.jar:23.4.0-dev-d54a1bf2]
	at org.hyperledger.besu.ethereum.permissioning.node.NodePermissioningController.isPermitted(NodePermissioningController.java:98) ~[besu-permissioning-23.4.0-SNAPSHOT.jar:23.4.0-dev-d54a1bf2]
	at org.hyperledger.besu.ethereum.permissioning.node.PeerPermissionsAdapter.inboundIsPermitted(PeerPermissionsAdapter.java:122) ~[besu-permissioning-23.4.0-SNAPSHOT.jar:23.4.0-dev-d54a1bf2]
	at org.hyperledger.besu.ethereum.permissioning.node.PeerPermissionsAdapter.isPermitted(PeerPermissionsAdapter.java:78) ~[besu-permissioning-23.4.0-SNAPSHOT.jar:23.4.0-dev-d54a1bf2]
	at org.hyperledger.besu.ethereum.p2p.permissions.PeerPermissions$CombinedPeerPermissions.isPermitted(PeerPermissions.java:129) ~[besu-p2p-23.4.0-SNAPSHOT.jar:23.4.0-dev-d54a1bf2]
	at org.hyperledger.besu.ethereum.p2p.rlpx.connections.PeerRlpxPermissions.allowOngoingConnection(PeerRlpxPermissions.java:61) ~[besu-p2p-23.4.0-SNAPSHOT.jar:23.4.0-dev-d54a1bf2]
	at org.hyperledger.besu.ethereum.p2p.rlpx.RlpxAgent.lambda$handlePermissionsUpdate$10(RlpxAgent.java:285) ~[besu-p2p-23.4.0-SNAPSHOT.jar:23.4.0-dev-d54a1bf2]
	at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183) ~[?:?]
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:179) ~[?:?]
	at java.util.stream.DistinctOps$1$2.accept(DistinctOps.java:174) ~[?:?]
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197) ~[?:?]
	at java.util.concurrent.ConcurrentHashMap$ValueSpliterator.forEachRemaining(ConcurrentHashMap.java:3612) ~[?:?]
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509) ~[?:?]
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499) ~[?:?]
	at java.util.stream.StreamSpliterators$WrappingSpliterator.forEachRemaining(StreamSpliterators.java:310) ~[?:?]
	at java.util.stream.Streams$ConcatSpliterator.forEachRemaining(Streams.java:734) ~[?:?]
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509) ~[?:?]
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499) ~[?:?]
	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150) ~[?:?]
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173) ~[?:?]
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[?:?]
	at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596) ~[?:?]
	at org.hyperledger.besu.ethereum.p2p.rlpx.RlpxAgent.handlePermissionsUpdate(RlpxAgent.java:283) ~[besu-p2p-23.4.0-SNAPSHOT.jar:23.4.0-dev-d54a1bf2]
	at org.hyperledger.besu.ethereum.p2p.permissions.PeerPermissions.lambda$dispatchUpdate$0(PeerPermissions.java:79) ~[besu-p2p-23.4.0-SNAPSHOT.jar:23.4.0-dev-d54a1bf2]
	at org.hyperledger.besu.util.Subscribers.lambda$forEach$0(Subscribers.java:131) ~[besu-util-23.4.0-SNAPSHOT.jar:23.4.0-dev-d54a1bf2]
	at java.lang.Iterable.forEach(Iterable.java:75) ~[?:?]
	at org.hyperledger.besu.util.Subscribers.forEach(Subscribers.java:128) ~[besu-util-23.4.0-SNAPSHOT.jar:23.4.0-dev-d54a1bf2]
	at org.hyperledger.besu.ethereum.p2p.permissions.PeerPermissions.dispatchUpdate(PeerPermissions.java:79) ~[besu-p2p-23.4.0-SNAPSHOT.jar:23.4.0-dev-d54a1bf2]
	at org.hyperledger.besu.ethereum.permissioning.node.PeerPermissionsAdapter.lambda$new$0(PeerPermissionsAdapter.java:54) ~[besu-permissioning-23.4.0-SNAPSHOT.jar:23.4.0-dev-d54a1bf2]
	at org.hyperledger.besu.ethereum.chain.DefaultBlockchain.lambda$appendBlockHelper$9(DefaultBlockchain.java:336) ~[besu-core-23.4.0-SNAPSHOT.jar:23.4.0-dev-d54a1bf2]
	at org.hyperledger.besu.util.Subscribers.lambda$forEach$0(Subscribers.java:131) ~[besu-util-23.4.0-SNAPSHOT.jar:23.4.0-dev-d54a1bf2]
	at java.lang.Iterable.forEach(Iterable.java:75) ~[?:?]
	at org.hyperledger.besu.util.Subscribers.forEach(Subscribers.java:128) ~[besu-util-23.4.0-SNAPSHOT.jar:23.4.0-dev-d54a1bf2]
	at org.hyperledger.besu.ethereum.chain.DefaultBlockchain.appendBlockHelper(DefaultBlockchain.java:336) ~[besu-core-23.4.0-SNAPSHOT.jar:23.4.0-dev-d54a1bf2]
	at org.hyperledger.besu.ethereum.chain.DefaultBlockchain.appendBlock(DefaultBlockchain.java:286) ~[besu-core-23.4.0-SNAPSHOT.jar:23.4.0-dev-d54a1bf2]
	at org.hyperledger.besu.ethereum.mainnet.MainnetBlockImporter.lambda$importBlock$0(MainnetBlockImporter.java:53) ~[besu-core-23.4.0-SNAPSHOT.jar:23.4.0-dev-d54a1bf2]
	at java.util.Optional.ifPresent(Optional.java:178) ~[?:?]
	at org.hyperledger.besu.ethereum.mainnet.MainnetBlockImporter.importBlock(MainnetBlockImporter.java:51) ~[besu-core-23.4.0-SNAPSHOT.jar:23.4.0-dev-d54a1bf2]
	at org.hyperledger.besu.ethereum.core.BlockImporter.importBlock(BlockImporter.java:45) ~[besu-core-23.4.0-SNAPSHOT.jar:23.4.0-dev-d54a1bf2]
	at org.hyperledger.besu.consensus.ibft.statemachine.IbftRound.importBlockToChain(IbftRound.java:316) ~[besu-ibft-23.4.0-SNAPSHOT.jar:23.4.0-dev-d54a1bf2]
	at org.hyperledger.besu.consensus.ibft.statemachine.IbftRound.peerIsCommitted(IbftRound.java:289) ~[besu-ibft-23.4.0-SNAPSHOT.jar:23.4.0-dev-d54a1bf2]
	at org.hyperledger.besu.consensus.ibft.statemachine.IbftRound.handleCommitMessage(IbftRound.java:213) ~[besu-ibft-23.4.0-SNAPSHOT.jar:23.4.0-dev-d54a1bf2]
	at org.hyperledger.besu.consensus.ibft.statemachine.IbftBlockHeightManager.actionOrBufferMessage(IbftBlockHeightManager.java:214) ~[besu-ibft-23.4.0-SNAPSHOT.jar:23.4.0-dev-d54a1bf2]
	at org.hyperledger.besu.consensus.ibft.statemachine.IbftBlockHeightManager.handleCommitPayload(IbftBlockHeightManager.java:204) ~[besu-ibft-23.4.0-SNAPSHOT.jar:23.4.0-dev-d54a1bf2]
	at org.hyperledger.besu.consensus.common.bft.statemachine.BaseBftController.consumeMessage(BaseBftController.java:126) ~[besu-consensus-common-23.4.0-SNAPSHOT.jar:23.4.0-dev-d54a1bf2]
	at org.hyperledger.besu.consensus.ibft.statemachine.IbftController.handleMessage(IbftController.java:90) ~[besu-ibft-23.4.0-SNAPSHOT.jar:23.4.0-dev-d54a1bf2]
	at org.hyperledger.besu.consensus.common.bft.statemachine.BaseBftController.handleMessageEvent(BaseBftController.java:88) ~[besu-consensus-common-23.4.0-SNAPSHOT.jar:23.4.0-dev-d54a1bf2]
	at org.hyperledger.besu.consensus.common.bft.EventMultiplexer.handleBftEvent(EventMultiplexer.java:53) ~[besu-consensus-common-23.4.0-SNAPSHOT.jar:23.4.0-dev-d54a1bf2]
	at java.util.Optional.ifPresent(Optional.java:178) [?:?]
	at org.hyperledger.besu.consensus.common.bft.BftProcessor.run(BftProcessor.java:65) [besu-consensus-common-23.4.0-SNAPSHOT.jar:23.4.0-dev-d54a1bf2]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) [?:?]
	at java.lang.Thread.run(Thread.java:833) [?:?]
Caused by: java.lang.NullPointerException
	at java.util.Objects.requireNonNull(Objects.java:208) ~[?:?]
	at java.util.ImmutableCollections.listFromTrustedArray(ImmutableCollections.java:213) ~[?:?]
	at java.util.List.of(List.java:847) ~[?:?]
	at org.hyperledger.besu.ethereum.permissioning.NodeSmartContractV2PermissioningController.createPayload(NodeSmartContractV2PermissioningController.java:106) ~[besu-permissioning-23.4.0-SNAPSHOT.jar:23.4.0-dev-d54a1bf2]
	... 56 more
```

### Versions
* Software version: 23.4.0-dev-d54a1bf2
* Java version: openjdk-java-17
* OS Name & Version: Ubuntu 22.04.2 LTS
* Kernel Version: 5.19.0-38-generic
* Docker Version: 23.0.4

### Additional Information (Add any of the following or anything else that may be relevant)

Fragment of docker-compose.yml
```
...
x-defaults:

  logging: &logging-default
    options:
      max-size: '10m'
      max-file: '5'
    driver: json-file

  localization: &localization-default
    TZ: America/Sao_Paulo
    LANG: en_US.UTF-8
...
    image: ${IMAGE_BESU}
    logging: *logging-default
    environment:
      << : *localization-default
      LOG4J_CONFIGURATION_FILE: "/var/lib/besu/log.xml"
      BESU_DATA_PATH: "/var/lib/besu"
      BESU_GENESIS_FILE: "/var/lib/besu/genesis.json"
      BESU_STATIC_NODES_FILE: "/var/lib/besu/static-nodes.json"
      BESU_MAX_PEERS: "50"
      BESU_REMOTE_CONNECTIONS_LIMIT_ENABLED: "false"
      BESU_DISCOVERY_ENABLED: "true"
      BESU_MIN_GAS_PRICE: "0"
      BESU_P2P_PORT: "60606"
      BESU_RPC_HTTP_ENABLED: "true"
      BESU_RPC_HTTP_API: "ADMIN,ETH,TXPOOL,NET,IBFT,WEB3,DEBUG,TRACE,PERM"
      BESU_RPC_HTTP_CORS_ORIGINS: "*"
      BESU_HOST_ALLOWLIST: "*"
      BESU_METRICS_ENABLED: "true"
      BESU_METRICS_HOST: "0.0.0.0"

    command: >-
      --permissions-nodes-contract-enabled
      --permissions-nodes-contract-address "0xB65EA7377ad0cAffE2f9e5aaA7045c05334ecAc4"
      --permissions-nodes-contract-version=2
...
```

--- a/ethereum/permissioning/src/main/java/org/hyperledger/besu/ethereum/permissioning/NodeSmartContractV2PermissioningController.java
+++ b/ethereum/permissioning/src/main/java/org/hyperledger/besu/ethereum/permissioning/NodeSmartContractV2PermissioningController.java
@@ -96,14 +96,15 @@ public class NodeSmartContractV2PermissioningController
   private Bytes createPayload(final EnodeURL enodeUrl) {
     try {
       final String hexNodeIdString = enodeUrl.getNodeId().toUnprefixedHexString();
-      final String address = enodeUrl.toURI().getHost();
+      final String host = enodeUrl.toURI().getHost();
+      final String address = host == null ? enodeUrl.getIpAsString() : host;
       final int port = enodeUrl.getListeningPortOrZero();
 
       final Function connectionAllowedFunction =
           FunctionEncoder.makeFunction(
               "connectionAllowed",
               List.of("string", "string", "uint16"),
-              List.of(hexNodeIdString, address, port),
+              List.of(hexNodeIdString, address == null ? "" : address, port),
               List.of(Bool.TYPE_NAME));
       return Bytes.fromHexString(FunctionEncoder.encode(connectionAllowedFunction));
     } catch (Exception e) {

